### ISSUE TITLE
Migrate Datadog config (dashboards, APM, logging etc) to FedRamped instance 

### ISSUE THREAD

**Issue by ElijahLynn** [department-of-veterans-affairs/va.gov-cms#9540]
We've been [notified](https://dsva.slack.com/archives/CJYRZK2HH/p1655494972231319) by Platform-Infrastructure that on June 30th our Datadog account at http://datadoghq.com/ will be no more and move to http://vagov.ddog-gov.com/. 

See https://github.com/department-of-veterans-affairs/va.gov-team/issues/43094 for upstream ticket. 

We need to migrate all of our configuration: dashboards, APM, logging etc.

The Datadog team has provided migration scripts that we can use, and Platform-Infrastructure is available to assist us with the migration too. See > https://vfs.atlassian.net/wiki/spaces/OT/pages/2236416007/Migrate+Datadog+to+DDog-Gov.

## Acceptance Criteria
- [x] All existing CMS managed configuration from http://datadoghq.com/ now exists at http://vagov.ddog-gov.com/
- [x] Documentation is checked and updated if needed
- [x] Pagerduty integration with synthetic alerts still works

## Implementation
- consider the content-build metrics that currently use Datadog API, i.e. interval calculations
- consider integrations with PagerDuty

## Task List
- [x] User account migration
  - [x] Platform-CMS-DevOps team Datadog accounts (Okta)
  - [x] Non-CMS-DevOps existing CMS related accounts
    - https://github.com/department-of-veterans-affairs/va.gov-cms/issues/9540#issuecomment-1175682576
  - [x] Broader CMS team (not needed right now, they don't have access right now, but should after migration, new story needed)
    - https://github.com/department-of-veterans-affairs/va.gov-cms/issues/9709
- [x] Get list of all CMS managed configuration
- [x] Migrate
  - [x] Synthetics
    - [x] Terraform managed
    - [x] Non-Terraform managed (Lowell)
  - [x] APM - In progress (dual shipping)
    - https://github.com/department-of-veterans-affairs/va.gov-team/issues/42611
    - APM dual shipping is working but need to update our code/make PR
    - https://dsva.slack.com/archives/CJYRZK2HH/p1656698094173459?thread_ts=1655494972.231319&cid=CJYRZK2HH
    - https://github.com/department-of-veterans-affairs/devops/pull/11561
  - [x] Dashboards
    - [x] [PLATFORM-CMS] Prod CMS
    - [x] [CMS-TIC] CMS TIC Monitoring Dashboard
    - [x] [Platform CMS] 502s (Prod, Staging, Tugboat)
  - [x] Metrics pushes from GHA workflows 
    - https://github.com/department-of-veterans-affairs/content-build/pull/1198
    - https://github.com/department-of-veterans-affairs/content-build/pull/1199
    - https://github.com/department-of-veterans-affairs/content-build/pull/1202 
  - [x] Metrics collector in Drupal (node count, users etc)
    - https://github.com/department-of-veterans-affairs/va.gov-cms/pull/9648
  - [x] S3 ELB Logs to DD w/Cloudformation
    - https://github.com/department-of-veterans-affairs/devops/pull/11492
    - waiting to apply, my permissions got removed somehow https://dsva.slack.com/archives/CJYRZK2HH/p1656714861621069?thread_ts=1656711767.549559&cid=CJYRZK2HH
  - [x] Service Level Objectives
	  - [x] 30d and 90d
  - [x] Monitors (14)
    - [x] Terraform
    - [x] Non-TF - Sync (3)
      - https://dsva.slack.com/archives/C01U8EYHSEM/p1657748951400739
      - CMS Drupal Error Monitor
        - Update Sentry to send to new endpoint
      - CMS content release monitor
      - non-200 HTTP Status Code for POST /simplesaml/module.php/saml/sp/saml2-acs.php/default-sp
- [x] Application log forwarding not coming in (ELB w/Cloudformation is though), might be endpoint issue
  - https://github.com/department-of-veterans-affairs/devops/pull/11561
- [x] prod.cms.va.gov/metrics not showing up on Dashboard
- [x] Testing: 
  - [x] Pagerduty integration still works
- [x] Communications: 
  - [x] Notify VFS-CMS team of new URL
  - https://dsva.slack.com/archives/CT4GZBM8F/p1657844564427719
- [x] Update any Slack channel bookmarks
- [x] Update any documentation
  - https://github.com/department-of-veterans-affairs/va.gov-cms/pull/9648
  - https://vfs.atlassian.net/wiki/spaces/PCMS/pages/2159640605/View+Datadog+APM+Traces
  - https://vfs.atlassian.net/wiki/spaces/PCMS/pages/2126610488/Configure+Datadog+APM+for+Local+and+CI+Environments
  - https://vfs.atlassian.net/wiki/spaces/PCMS/pages/2035941389/Content+Build+Metrics+Dashboard+in+Datadog
  - https://github.com/department-of-veterans-affairs/va.gov-cms/blob/b81441c407ef0d5aa8c0c49c0e6c5ee7f5cd1dc6/.tugboat/config.yml
  - https://github.com/department-of-veterans-affairs/va.gov-cms/blob/b81441c407ef0d5aa8c0c49c0e6c5ee7f5cd1dc6/READMES/upstream-dependencies.md
  - https://github.com/department-of-veterans-affairs/va.gov-cms/blob/b81441c407ef0d5aa8c0c49c0e6c5ee7f5cd1dc6/docroot/modules/custom/va_gov_backend/src/Service/Datadog.php
  - https://github.com/department-of-veterans-affairs/va.gov-cms/blob/b81441c407ef0d5aa8c0c49c0e6c5ee7f5cd1dc6/READMES/downstream_dependencies.md

**Comment by cweagans** [comment]
Release interval calculation is here: https://github.com/department-of-veterans-affairs/content-build/blob/main/.github/workflows/record-release-interval/action.yml

This will be a little tricky. Probably worth talking through.

**Comment by olivereri** [comment]
Oseas Moran mentioned that what we owned in the datadog terraform environment has been migrated but some lingering resources remain:

https://dsva.slack.com/archives/CT4GZBM8F/p1655911951470539

**Comment by ElijahLynn** [comment]
> migrated but some lingering resources remain:

I've updated the task list to reflect the remaining resources. Synthetics were migrated, APM is [in progress](https://github.com/department-of-veterans-affairs/va.gov-team/issues/42611), dashboards, the ALB logs custom DD metrics in a GHA remain.

**Comment by ElijahLynn** [comment]
Some DD monitors have the old Slack group @slack-cms-notifications, and need to be updated to use @cms-alerts-medium.

**Comment by ElijahLynn** [comment]
I've requested access for the following users who had access on the old instance who don't have it on the new instance:

https://vajira.max.gov/servicedesk/customer/portal/1/DOTSD-16696

Aaron Ponce	https://github.com/amponce
Clarence Maeng	https://github.com/cmaeng
Cameron Eagans	https://github.com/cweagans
Daniel Sasser	https://github.com/dsasser
Elizabeth Neal	https://github.com/ENeal49
Jessica Alexander	https://github.com/jkalexander7
Lori Pusey	https://github.com/loripusey
Lynn Stahl	https://github.com/msbtterswrth
Nira Datta	https://github.com/niradatta1
Steve Wirt	https://github.com/swirtSJW
Yaching Tsao	https://github.com/ytsaoca
Zach Park	https://github.com/zach-park

**Comment by olivereri** [comment]
Upstream issue:
https://github.com/DataDog/datadog-serverless-functions/issues/580

**Comment by ElijahLynn** [comment]
* PR open here to get APM and logs working https://github.com/department-of-veterans-affairs/devops/pull/11561
* Next PROD deploy should have the metrics working ingested (I had to set the correct API key in SSM, tested, worked)
* Pagerduty integration should be working now, per my request here to Infra > https://dsva.slack.com/archives/CJYRZK2HH/p1657665984632359, will test tomorrow.

**Comment by ElijahLynn** [comment]
K, today's update is that the metrics are working now and the APM and Logging will work on tomorrow's deploy (PR was delayed do to merging in the disabling of dual-shipping, which prevented some technical debt). The remaining work will be to port over 3 monitors and update the Sentry endpoint, which will happen tomorrow. I was blocked on porting the monitors off but figured out a way to see the config on the DD Commercial instance by using some JS to remove the paywall overlay and blur so I could see what they were and still reach the edit pages to grab the config, even though the paywall overlay is still up. See https://dsva.slack.com/archives/C01U8EYHSEM/p1657748951400739 for details. 

Here is the JS code:

```
document.querySelector(".single-page-app_paywall_paywall").remove()
document.querySelector("body").classList.remove("paywall")
 ```
 
 > Had to search for /monitor/ in the list view (/monitors?q=cms) to find the IDs and then the format is just /monitor/3455555/edit. Then run the JS and use the collapse feature to see everything (scroll is disabled and I didn't figure out how to enable that yet, and don't need to it appears).

**Comment by ElijahLynn** [comment]
All is working and migrated. I notified team in #cms-team channel here https://dsva.slack.com/archives/CT4GZBM8F/p1657844564427719 with this message. 

> :mega: Datadog migration is complete. We migrated from the DD Commercial instance at [app.datadoghq.com](http://app.datadoghq.com/) to the FedRAMPed instance at [vagov.ddog-gov.com](http://vagov.ddog-gov.com/). Please update any bookmarks you may have. The main PROD CMS dashboard is at https://vagov.ddog-gov.com/dashboard/vnk-g4s-fru/platform-cms-prod-cms and does have panel to show the current log stream as well as some metrics. This dashboard is not as complete as Grafana (https://grafana.vfs.va.gov/d/dxf8a-6Zz/cms-dashboard) yet but will eventually have similar metrics. And you can access the DD Dashboard off of SOCKS.
You will need an Okta account to access the DD GOV instance, existing accounts have already received invites from Okta.
If you don't have an Okta account yet you will be getting one in this ticket https://github.com/department-of-veterans-affairs/va.gov-cms/issues/9709 (subscribe if you like),
